src/wallet/walletdb.cpp

```diff
@@ -670,20 +670,7 @@ void MaybeCompactWalletDB()
     }
 
     for (const std::shared_ptr<CWallet>& pwallet : GetWallets()) {
```

Comment:
So ideally this would iterate over current wallet databases, not current wallets, it's just that we don't have an iterator like that. Note that it does nothing with `pwallet` eacept get its database.